pvlib/pvsystem.py

```diff
@@ -1223,6 +1255,146 @@ def calcparams_desoto(effective_irradiance, temp_cell,
+    a_ref : float
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at reference
+        conditions, in units of V.
+
+    I_L_ref : float
+        The light-generated current (or photocurrent) at reference conditions,
+        in amperes.
+
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions,
+        in amperes.
+
+    R_sh_ref : float
+        The shunt resistance at reference conditions, in ohms.
+
+    R_s : float
+        The series resistance at reference conditions, in ohms.
+
+    Adjust : float
+        The adjustment to the temperature coefficient for short circuit
+        current, in percent
+
+    EgRef : float
+        The energy bandgap at reference temperature in units of eV.
+        1.121 eV for crystalline silicon. EgRef must be >0.  For parameters
+        from the SAM CEC module database, EgRef=1.121 is implicit for all
+        cell types in the parameter estimation algorithm used by NREL.
+
+    dEgdT : float
+        The temperature dependence of the energy bandgap at reference
+        conditions in units of 1/K. May be either a scalar value
+        (e.g. -0.0002677 as in [3]) or a DataFrame (this may be useful if
+        dEgdT is a modeled as a function of temperature). For parameters from
+        the SAM CEC module database, dEgdT=-0.0002677 is implicit for all cell
+        types in the parameter estimation algorithm used by NREL.
+
+    irrad_ref : float (optional, default=1000)
+        Reference irradiance in W/m^2.
+
+    temp_ref : float (optional, default=25)
+        Reference cell temperature in C.
+
+    Returns
+    -------
+    Tuple of the following results:
+
+    photocurrent : numeric
+        Light-generated current in amperes
+
+    saturation_current : numeric
+        Diode saturation curent in amperes
+
+    resistance_series : float
+        Series resistance in ohms
+
+    resistance_shunt : numeric
+        Shunt resistance in ohms
+
+    nNsVth : numeric
+        The product of the usual diode ideality factor (n, unitless),
+        number of cells in series (Ns), and cell thermal voltage at
+        specified effective irradiance and cell temperature.
+
+    References
+    ----------
+    [1] A. Dobos, "An Improved Coefficient Calculator for the California
+    Energy Commission 6 Parameter Photovoltaic Module Model", Journal of
+    Solar Energy Engineering, vol 134, 2012.
+
+    [2] System Advisor Model web page. https://sam.nrel.gov.
+
+    [3] W. De Soto et al., "Improvement and validation of a model for
+    photovoltaic array performance", Solar Energy, vol 80, pp. 78-88,
+    2006.
+
+    See Also
+    --------
+    calcparams_desoto
+    singlediode
+    retrieve_sam
+
+    Notes
+    -----
+    The CEC model differs from the De soto et al. model [3] only in the
+    parameter adjust.
+    '''
+
+    # test for use of function pre-v0.6.0 API change
```

Comment:
Do we need to include this check? It's a new function, so I don't think we need to support an old API. Better to keep it simple if possible.

Reply:
But the ``ModelChain`` path will go through a ``PVSystem.calcparams_cec/desoto`` method rather than calling the function directly. And we updated the method to use the correct signature. So, I don't see how we could break anyone's code.